server/issue.go

```diff
@@ -310,6 +311,9 @@ func (s *Server) composeIssueRelationship(ctx context.Context, issue *api.Issue)
 }
 
 func (s *Server) createIssue(ctx context.Context, issueCreate *api.IssueCreate, creatorID int) (*api.Issue, error) {
+	if issueCreate.ValidateOnly {
```

Comment:
I am actually thinking to just pass an extra validateOnly flag to createIssue.

We need to skip CreatePipeline/Stage/Task/Issue and fake the ID, but we can still share most of the code and have a better chance to have a more realistic validation.

e.g. The current PR still has couple issues:

- 	would not catch the if issueCreate.AssigneeID == api.UnknownID {
		return nil, echo.NewHTTPError(http.StatusBadRequest, "Failed to create issue, assignee missing")
	}

-  The FindInstance check in the stateCreate.TaskList

wdyt?